component/file_cache/file_cache.go

```diff
@@ -442,8 +442,32 @@ func (fc *FileCache) DeleteDir(options internal.DeleteDirOptions) error {
 
 // StreamDir : Add local files to the list retrieved from storage container
 func (fc *FileCache) StreamDir(options internal.StreamDirOptions) ([]*internal.ObjAttr, string, error) {
+	// For stream directory, there are three different child path situations we have to potentially handle.
+	// 1. Path in storage but not in local cache
+	// 2. Path not in storage but in local cache (this could happen if we recently created the file [and are currently writing to it]) (also supports immutable containers)
+	// 3. Path in storage and in local cache (this could result in dirty properties on the service if we recently wrote to the file)
+
+	// To cover case 1, grab all entries from storage
 	attrs, token, err := fc.NextComponent().StreamDir(options)
 
+	for _, attr := range attrs {
```

Comment:
I think we optimize this by using a double-indexed loop, which walks results from `os.ReadDir` and `attrs` in parallel?
`attrs` is already alphabetized (I think!), and the `os.ReadDir` results can be alphabetized quickly using `slices.SortFunc`.
If you implement this suggestion, then since we call `os.ReadDir` to merge case 2 entries with the last chunk, maybe we could use a single `os.ReadDir` call and use the results for either thing based on whether we're on the last chunk.

Reply:
Yes, using os.ReadDir is faster than the stat. I used your suggestion of a double-indexed loop and it is faster on Linux compared to the stat method when the cache is small. However, once the cache is larger, then this method will be slower. It is about a .5 -1ms penalty for every 500 files in the cache. So as long as the number of items in a folder is less than ~2000, this is better. On Windows, performance is significantly improved, with about a 1-2ms penalty per 500 files.